pkg/api/handlers/libpod/images.go

```diff
@@ -462,31 +457,21 @@ func PushImage(w http.ResponseWriter, r *http.Request) {
 	defer auth.RemoveAuthfile(authfile)
 	logrus.Errorf("AuthConf: %v", authConf)
 
-	dockerRegistryOptions := &image.DockerRegistryOptions{
-		DockerRegistryCreds: authConf,
+	options := entities.ImagePushOptions{
+		Authfile: authfile,
+		Username: authConf.Username,
+		Password: authConf.Password,
+		Format:   query.Format,
 	}
 	if sys := runtime.SystemContext(); sys != nil {
-		dockerRegistryOptions.DockerCertPath = sys.DockerCertPath
-		dockerRegistryOptions.RegistriesConfPath = sys.SystemRegistriesConfPath
+		options.CertDir = sys.DockerCertPath
```

Comment:
RegistriesConfPath must be set as well

Reply:
Neither of these fields are supported from the remote side, so they both should be dropped and allow the server to handle it.  I beleive.